contracts/Depository.sol

```diff
@@ -65,13 +65,16 @@ contract Depository is IErrorsTokenomics {
     event TreasuryUpdated(address indexed treasury);
     event BondCalculatorUpdated(address indexed _bondCalculator);
     event CreateBond(address indexed token, uint256 indexed productId, address indexed owner, uint256 bondId,
-        uint256 amountOLAS, uint256 tokenAmount, uint256 expiry);
-    event RedeemBond(uint256 indexed productId, address indexed owner, uint256 bondId);
-    event CreateProduct(address indexed token, uint256 indexed productId, uint256 supply, uint256 priceLP, uint256 expiry);
-    event CloseProduct(address indexed token, uint256 indexed productId);
+        uint256 amountOLAS, uint256 tokenAmount, uint256 maturity, uint256 startTime);
+    event RedeemBond(uint256 indexed productId, address indexed owner, uint256 bondId, uint256 redeemTime);
+    event CreateProduct(address indexed token, uint256 indexed productId, uint256 supply, uint256 priceLP,
+        uint256 vesting, uint256 startTime);
+    event CloseProduct(address indexed token, uint256 indexed productId, uint256 closeTime);
 
     // Minimum bond vesting value
     uint256 public constant MIN_VESTING = 1 days;
+    // Maximum bond vesting value
+    uint256 public constant MAX_VESTING = 4 * 365 days;
```

Comment:
Why?

Reply:
We didn't have any limit for vesting before, we believe it must be set. Also, better to check for the overflow values with it.

Reply:
I think the natural limit is max of uint32? So I see that as a relevant bound. Not sure why 4 years has any relevance.

Reply:
Ok, let's revert this change.